source/config/Config.cpp

```diff
@@ -1012,6 +1040,17 @@ bool PlainConfig::PubSub::Validate() const
             DeviceClient::DC_FATAL_ERROR);
         return false;
     }
+    if (publishFile.has_value() && !publishFile->empty())
+    {
+        if (!FileUtils::ValidateFilePermissions(publishFile.value(), Permissions::PUB_SUB_FILES, true))
+        {
+            LOGM_ERROR(
+                Config::TAG,
+                "*** %s: Publish File field must be specified if Pub-Sub sample feature is enabled ***",
```

Comment:
This might be confusing because if we made it here, something was specified... it just failed the file permission check

Reply:
Noted. Will remove it.